Makefile.am

```diff
@@ -41,11 +41,15 @@
 # OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 #
 
+WARN_CFLAGS = -Wall -Wextra -Werror
+WARN_CPPFLAGS = -Wall -Wextra -Werror
+
 ACLOCAL_AMFLAGS = -I cmulocal
-AM_CFLAGS = @PERL_CCCDLFLAGS@ $(GCOV_CFLAGS)
+AM_CFLAGS = $(WARN_CFLAGS) @PERL_CCCDLFLAGS@ $(GCOV_CFLAGS)
 AM_CXXFLAGS = $(GCOV_CXXFLAGS)
```

Comment:
Any reason you didn't update `AM_CXXFLAGS`?  We do have C++ code in Cyrus (xapian_wrap.cpp, icu_wrap.cpp), and we want warnings-as-errors for those files too.

My own wrapper script already sets these options in `CXXFLAGS`, and the C++ code builds clean, for what it's worth.

Reply:
Agreed. I now set CXXFLAGS to CFLAGS, but locally I ran CXXFLAGS additionally with `-Wpedantic` for quite some time.